client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClientFactory.java

```diff
@@ -95,6 +95,26 @@ public void addReconnectListener(final ReconnectListener<HonoConnection> listene
         return connection.isConnected();
     }
 
+    /**
+     * Checks whether this client is connected to the service.
+     * <p>
+     * If a connection attempt is currently in progress, the returned future is completed
+     * with the outcome of the connection attempt. If the connection attempt (including
+     * potential reconnect attempts) isn't finished after the given timeout, the returned
+     * future is failed.
+     * <p>
+     * Simply delegates to {@link HonoConnection#isConnected(long)}.
+     * 
+     * @param waitForCurrentConnectAttemptTimeout The maximum number of milliseconds to wait for
+     *                                            on ongoing connection attempt to finish.
+     * @return A succeeded future if this factory is connected.
+     *         Otherwise, the future will be failed with a {@link ServerErrorException}.
+     */
+    @Override
+    public final Future<Void> isConnected(final long waitForCurrentConnectAttemptTimeout) {
+        return connection.isConnected();
```

Comment:
shouldn't this be `connection.isConnected(waitForCurrentConnectionAttemptTimeout)`?